src/cpp/r/session/RSession.cpp

```diff
@@ -633,15 +633,21 @@ FilePath tempFile(const std::string& prefix, const std::string& extension)
    return filePath;
 }
 
-FilePath tempDir()
+namespace {
+
+FilePath tempDirImpl()
 {
-   std::string tempDir;
-   Error error = r::exec::RFunction("tempdir").call(&tempDir);
-   if (error)
-      LOG_ERROR(error);
+   return FilePath(
+       string_utils::systemToUtf8(
+           r::util::fixPath(R_TempDir)));
+}
 
-   FilePath filePath(string_utils::systemToUtf8(r::util::fixPath(tempDir)));
-   return filePath;
+} // end anonymous namespace
+
+FilePath tempDir()
+{
+   static FilePath instance = tempDirImpl();
+   return instance;
```

Comment:
This change is unrelated to the PR but felt small + safe enough to include here. The `create_terminal` RPC can potentially be run in a separate R thread, but that uses this API here:

https://github.com/rstudio/rstudio/blob/f7c7573d469f3ce7b440f23b3906c6b6b72c2bd3/src/cpp/session/SessionConsoleProcess.cpp#L975-L977

This change makes this API thread-safe so it can be used safely in this scenario.

Note that the R `tempdir()` function effectively does the same thing, and this value is fixed once R has finished initialization.

https://github.com/wch/r-source/blob/4d6e1a77d487736f0d95ba8c8b71a8a527e55f22/src/main/sysutils.c#L250